src/Illuminate/Auth/Access/Grant.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Illuminate\Auth\Access;
+
+class Grant
+{
+    /**
+     * Indicates whether the request was granted.
+     *
+     * @var bool
+     */
+    protected $allowed;
+
+    /**
+     * The reason the request was granted/denied.
+     *
+     * @var string|null
+     */
+    protected $reason;
+
+    /**
+     * Create a new access grant instance.
+     *
+     * @return void
+     */
+    protected function __construct()
+    {
+        //
```

Comment:
maybe include a comment here regarding the significance of making this method protected - can be useful for less experienced php developers reading laravel's source